### PR TITLE
KFSPTS-19667 handling vendor contacts with a phone number correctly and handle PMW wire vendors

### PR DESCRIPTION
**Pull request by jhulslander**
Please note there is a related non prod SQL pr for this.

The wire related rules that I implemented are on this Jira: https://jira.cornell.edu/browse/KFSPTS-19667 .

### CODE REVIEW

File: src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java [hunk 1]
```diff
@@ -206,12 +216,175 @@ void testBuildPOFipsCountryCodeBadCountryForeignMode() {
...
+        PaymentWorksVendor pmwVendor = new PaymentWorksVendor();
+        pmwVendor.setRequestingCompanyTaxCountry(ARUBA_ISO_CODE);
+        pmwVendor.setPaymentMethod(StringUtils.EMPTY);
+        try {
+            conversionService.buildVendorDetailExtension(pmwVendor,  buildPaymentWorksIsoToFipsCountryMap());
+        } catch (IllegalArgumentException iae) {
+            assertTrue(true, "expected illegal argument exception");
+            return;
+        }
+        assertTrue(false, "expected illegal argument exception");
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java:343]: You could simplify this by using JUnit's fail() assertion instead, and place the assertion within the "try" block.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a few areas that could use some minor revisions, but otherwise the code changes look good.

Also, it looks like several unit tests failed, but not because of these changes. Some of our integration tests may need to be updated to account for any recent UTEST data changes. We should create a separate user story to fix the affected tests.

File: src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java [hunk 2]
```diff
@@ -13,24 +13,32 @@
...
     private PaymentWorksBatchReportSummaryItem recordsWithForeignAchSummary;
+    private PaymentWorksBatchReportSummaryItem recordsWithPaymentMethodWireDomesticSummary;
+    private PaymentWorksBatchReportSummaryItem recordsWithPaymentMethodWireForeignSummary;
     
     private List<PaymentWorksBatchReportVendorItem> disapprovedVendors;
     private List<PaymentWorksBatchReportVendorItem> noAchDataProvidedVendors;
     private List<PaymentWorksBatchReportVendorItem> pmwVendorAchsThatCouldNotBeProcessed;
     private List<PaymentWorksBatchReportVendorItem> recordsGeneratingException;
     private List<PaymentWorksBatchReportVendorItem> foreignAchItems;
+    private List<PaymentWorksBatchReportVendorItem> paymentMethodWireDomsticItems;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java:24]: ok

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorPayeeAchServiceImpl.java [hunk 3]
```diff
@@ -299,6 +300,42 @@ protected boolean isUsAchBank(PaymentWorksVendor pmwVendor, PaymentWorksNewVendo
...
+                                errorMessages));
+                    } else {
+                        LOG.debug("isAchPaymentMethod, found a foreign WIRE payment method, return false.");
+                        reportData.getRecordsWithPaymentMethodWireForeignSummary().incrementRecordCount();
+                        errorMessages.add(configurationService.getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_PAYMENTWORKS_FOREIGN_WIRE_VENDOR_ACH));
+                        reportData.addPaymentWireForeignItem(getPaymentWorksNewVendorPayeeAchReportService().createBatchReportVendorItem(pmwVendor, 
+                                errorMessages));
+                    }
+                } else {
+                    LOG.error("isAchPaymentMethod, We should only see wire or ACH or wire payment methods hitting this function, found an unexpected method: "
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorPayeeAchServiceImpl.java:329]: yes, good catch

File: src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java [hunk 4]
```diff
@@ -7,33 +7,43 @@
...
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.core.config.Configurator;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.vnd.businessobject.VendorContact;
+import org.kuali.kfs.vnd.businessobject.VendorContactPhoneNumber;
+import org.kuali.rice.core.impl.datetime.DateTimeServiceImpl;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java:20]: ok

File: src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java [hunk 5]
```diff
@@ -206,12 +216,175 @@ void testBuildPOFipsCountryCodeBadCountryForeignMode() {
...
+                CONTACT_PHONE_NUMBER, CONTACT_PHONE_NUMBER_EXTENSION);
+        assertBaseContactDetails(actualContact);
+        assertEquals(1, CollectionUtils.size(actualContact.getVendorContactPhoneNumbers()));
+        
+        VendorContactPhoneNumber actualPhoneContact = actualContact.getVendorContactPhoneNumbers().get(0);
+        assertEquals(PaymentWorksConstants.KFSVendorContactPhoneTypes.E_INVOICING, actualPhoneContact.getVendorPhoneTypeCode());
+        assertEquals(CONTACT_PHONE_NUMBER, actualPhoneContact.getVendorPhoneNumber());
+        assertEquals(CONTACT_PHONE_NUMBER_EXTENSION, actualPhoneContact.getVendorPhoneExtensionNumber());
+    }
+    @Test
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java:245]: ok

File: src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java (hunk 1 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImplTest.java]: OK, I switched

**Comment by jhulslander** [comment]
I have made some changes based on the code review, this is ready for review again

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed a couple very minor follow-up spots that could use a quick cleanup, but otherwise the changes look good.

File: src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java [hunk 6]
```diff
@@ -156,5 +164,51 @@ public void addForeignAchItem(PaymentWorksBatchReportVendorItem foreignAchItem)
...
+
+    public void setRecordsWithPaymentMethodWireForeignSummary(PaymentWorksBatchReportSummaryItem recordsWithPaymentMethodWireForeignSummary) {
+        this.recordsWithPaymentMethodWireForeignSummary = recordsWithPaymentMethodWireForeignSummary;
+    }
+    
+    public List<PaymentWorksBatchReportVendorItem> getPaymentMethodWireDomesticItems() {
+        return paymentMethodWireDomesticItems;
+    }
+
+    public void setPaymentMethodWireDomesticItems(List<PaymentWorksBatchReportVendorItem> paymentMethodWireDomsticItems) {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java:188]: ok

[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java:188]: ok

File: src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java [hunk 7]
```diff
@@ -156,5 +164,51 @@ public void addForeignAchItem(PaymentWorksBatchReportVendorItem foreignAchItem)
...
+        this.paymentMethodWireDomesticItems = paymentMethodWireDomsticItems;
+    }
+    
+    public void addPaymentWireDomesticItem(PaymentWorksBatchReportVendorItem paymentMethodWireDomesticItems) {
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/pmw/batch/report/PaymentWorksNewVendorPayeeAchBatchReportData.java:192]: ok

**Comment by jhulslander** [comment]
OK, I fixed the items noted, this is ready for review again

**Review by cah292 — APPROVED** [review]
Changes look good, and the only failed unit tests are those from the known KFSPTS-19792 problem and not from these PMW changes. The related SQL changes are also ready.

Merging.

**Pull request merged**